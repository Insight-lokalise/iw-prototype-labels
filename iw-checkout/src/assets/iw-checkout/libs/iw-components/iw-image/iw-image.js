import React from 'react'
/**
 * A component that wraps a <img> tag.
 * Enforces an alt attribute.
 *
 * We could do a few more things with this:
 *  - Lazy load
 *  - Display a loading icon while the image is loading
 *
 * @param {props} { src, alt, ...props } src/alt are required.
 */
export function IWImage({ src, alt, ...props }) {
    if (!alt) throw Error('Insight Images must have an alt-text attribute.')
    if (!src) src = 'https://prod-assets.insight.com/ccms_img/noImageAvailable_150x112.png'
    return (
        <img className={props.className}
            src={src}
            alt={alt}
            onClick={props.onClick}
            { ...props } />
    )
}
