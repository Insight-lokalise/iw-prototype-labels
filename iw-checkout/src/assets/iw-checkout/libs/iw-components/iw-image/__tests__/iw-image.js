import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { IWImage } from './../iw-image'

describe('IWImage', () => {
    it('should throw if alt-text is not provided', () => {
        expect(() => renderer.create(<IWImage />)).toThrow(/Images must have an alt-text/)
    })

    it('should default src if none is provided', () => {
        const component = shallow(
            <IWImage alt='sm' />
        )
        expect(component.prop('src')).toBe('https://prod-assets.insight.com/ccms_img/noImageAvailable_150x112.png')
    })

    it('should match the previous snapshot', () => {
        const component = renderer.create(
            <IWImage alt='sm' custom='prop' />
        )
        expect(component).toMatchSnapshot()
    })
})
