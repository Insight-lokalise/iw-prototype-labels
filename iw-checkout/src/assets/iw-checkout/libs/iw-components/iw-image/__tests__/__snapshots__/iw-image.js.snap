// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IWImage should match the previous snapshot 1`] = `
<img
  alt="sm"
  className={undefined}
  custom="prop"
  onClick={undefined}
  src="https://prod-assets.insight.com/ccms_img/noImageAvailable_150x112.png"
/>
`;
